tests/unit/test_hookutils.py

```diff
@@ -186,3 +186,45 @@ def test_deprecated_upstart_functions(self) -> None:
             apport.hookutils.attach_upstart_logs({}, "apport")
         with self.assertWarns(PendingDeprecationWarning):
             apport.hookutils.attach_upstart_overrides({}, "nonexisting")
+
+    def test_recent_syslog_race_condition(self) -> None:
+        """Test recent_syslog is not racing process completion.
+
+        LP: #2073935
+        Test read buffer after process completion.
+        """
+        with unittest.mock.patch("apport.hookutils.subprocess.Popen") as popen:
+            process = unittest.mock.Mock()
+            popen.return_value.__enter__.return_value = process
+
+            process.poll = unittest.mock.Mock(return_value=0)
+            process.returncode = 0
+            buffer_mock = unittest.mock.MagicMock()
+            buffer_mock.__iter__.side_effect = [iter([b"content"])]
+            process.stdout = buffer_mock
```

Comment:
idea: Maybe that is more readable?
```suggestion
            process.stdout = unittest.mock.MagicMock()
            process.stdout.__iter__.side_effect = [iter([b"content"])]
```